qiskit_nature/properties/second_quantization/electronic/electronic_energy.py

```diff
@@ -118,29 +128,96 @@ def from_driver_result(cls, result: DriverResult) -> "ElectronicEnergy":
         Raises:
             QiskitNatureError: if a WatsonHamiltonian is provided.
         """
-        cls._validate_input_type(result, ElectronicDriverResult)
+        cls._validate_input_type(result, LegacyElectronicDriverResult)
 
         qmol = cast(QMolecule, result)
 
         energy_shift = qmol.energy_shift.copy()
-        energy_shift["nuclear repulsion"] = qmol.nuclear_repulsion_energy
 
-        return cls(
-            ElectronicBasis.MO,
-            {
-                1: OneBodyElectronicIntegrals(
+        integrals: List[ElectronicIntegrals] = []
+        if qmol.hcore is not None:
+            integrals.append(
+                OneBodyElectronicIntegrals(ElectronicBasis.AO, (qmol.hcore, qmol.hcore_b))
+            )
+        if qmol.eri is not None:
+            integrals.append(
+                TwoBodyElectronicIntegrals(ElectronicBasis.AO, (qmol.eri, None, None, None))
+            )
+        if qmol.mo_onee_ints is not None:
+            integrals.append(
+                OneBodyElectronicIntegrals(
                     ElectronicBasis.MO, (qmol.mo_onee_ints, qmol.mo_onee_ints_b)
-                ),
-                2: TwoBodyElectronicIntegrals(
+                )
+            )
+        if qmol.mo_eri_ints is not None:
+            integrals.append(
+                TwoBodyElectronicIntegrals(
                     ElectronicBasis.MO,
-                    (
-                        qmol.mo_eri_ints,
-                        qmol.mo_eri_ints_ba,
-                        qmol.mo_eri_ints_bb,
-                        None,
-                    ),
-                ),
-            },
+                    (qmol.mo_eri_ints, qmol.mo_eri_ints_ba, qmol.mo_eri_ints_bb, None),
+                )
+            )
+
+        ret = cls(
+            integrals,
             energy_shift=energy_shift,
+            nuclear_repulsion_energy=qmol.nuclear_repulsion_energy,
             reference_energy=qmol.hf_energy,
         )
+
+        orb_energies = qmol.orbital_energies
+        if qmol.orbital_energies_b is not None:
+            orb_energies = np.asarray((qmol.orbital_energies, qmol.orbital_energies_b))
+        ret.orbital_energies = orb_energies
+
+        if qmol.kinetic is not None:
+            ret.kinetic = OneBodyElectronicIntegrals(ElectronicBasis.AO, (qmol.kinetic, None))
+
+        if qmol.overlap is not None:
+            ret.overlap = OneBodyElectronicIntegrals(ElectronicBasis.AO, (qmol.overlap, None))
+
+        return ret
+
+    def matrix_operator(self, density: OneBodyElectronicIntegrals) -> OneBodyElectronicIntegrals:
```

Comment:
It is a bit difficult to find a concise name. This method gets defined by the `IntegralProperty` class. In the case of the `ElectronicEnergy` it essentially computes the Fock operator (or the Hamiltonian) but in the case of e.g. the dipole moments, this operator is simply the one-electron integrals because they don't take any 2-electron contributions.

How about `integral_operator`? Although that is also still rather general (but that is kind of the point of this method). The idea is, "given an electronic density (see below) compute the operator describing this property".

A note on the density: right now, this is a simple `OneBodyElectronicIntegrals` instance, but in the future this may likely become its own `Property` instead.

Reply:
I guess matrix_operator gets blurred in my mind with matrix operator in Terra (opflow and quantum info) which is the same operator but in matrix form - hence I was looking for some more about what this is, which is about more what it contains than the representation of that operator (eg sum of paulis versus matrix). Since we have the FermionicOp etc that are all operators I wonder whether others will be confused or not. I see you have method `fock` which states in the description that this is the fock operator. Otherwise this seems to be the only method I noticed with operator in the name and it also talks about building the Fock operator as its docstring; but presumbly this is not be be confused with 'fock' method.

Reply:
Yes the `fock` property is more a preparational addition for the DFT Embedding. I think I can remove that again for now to avoid confusion. Indeed, this will be related to the `integral_operator` method.